client.ts

```diff
@@ -0,0 +1,463 @@
+  }
+
+  /**
+   * Expand - Expands the leaves that have a particular relation to an object (evaluates)
+   * @param expandRequest
+   * @param {string} expandRequest.relation The relation
+   * @param {string} expandRequest.object The object, must be of the form: `<type>:<id>`
+   * @param options
+   */
+  async expand(expandRequest: ClientExpandRequest, options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ExpandResponse> {
+    return this.openFgaApi.expand({
+      authorization_model_id: this.getAuthorizationModelId(options),
+      tuple_key: expandRequest,
+    });
+  }
+
+  /**
+   * ListObjects - List the objects of a particular type that the user has a certain relation to (evaluates)
+   * @param listObjectsRequest
+   * @param options
+   */
+  async listObjects(listObjectsRequest: ClientListObjectsRequest, options: ClientRequestOptsWithAuthZModelId = {}): PromiseResult<ListObjectsResponse> {
+    const authorizationModelId = this.getAuthorizationModelId(options);
+    return this.openFgaApi.listObjects({
+      authorization_model_id: authorizationModelId,
+      user: listObjectsRequest.user,
+      relation: listObjectsRequest.relation,
+      type: listObjectsRequest.type,
+      contextual_tuples: { tuple_keys: listObjectsRequest.contextual_tuples || [] },
+    }, options);
+  }
+
+  /**
+   * ListRelations - Lists all the relations a user has with an object (evaluates)
+   * @param {object} listRelationsRequest
+   * @param {string} listRelationsRequest.user The user object, must be of the form: `<type>:<id>`
+   * @param {string} listRelationsRequest.object The object, must be of the form: `<type>:<id>`
+   * @param {string[]} [listRelationsRequest.relations] The list of relations to check, if not sent default to the all those for that type int the model
+   * @param options
+   */
+  async listRelations(listRelationsRequest: ClientListRelationsRequest, options: ClientRequestOptsWithAuthZModelId = {}): Promise<ClientListRelationsResponse> {
+    const { user, object } = listRelationsRequest;
+    let { relations } = listRelationsRequest;
+    
+    // 1- Get the list of relations on that object type
+    // 2- Filter out/error out on invalid relations
+    // 2- Call check in batch
+    const authorizationModel: AuthorizationModel | undefined = this.getAuthorizationModelId(options) ?
+      (await this.readAuthorizationModel()).authorization_model : (await this.readLatestAuthorizationModel(options)).authorization_model;
+
+    if (!authorizationModel) {
+      throw new Error("authorization_model_not_found");
+    }
+
+    const { type: objectType } = getObjectFromString(object);
+    const availableRelations = Object.keys(authorizationModel?.type_definitions?.find(typeDef => typeDef.type === objectType)?.relations || {});
+
+    if (!relations) {
+      relations = availableRelations;
+    }
+
+    if (relations.some(relation => !availableRelations.includes(relation))) {
```

Comment:
I am a bit iffy about needing to read the authorization model itself and validate whether such model is possible in the case relations are specified.

Will it make more sense to rely on the check to throw relations not defined exception?  This reduces the need for extra reading of auth model in the common (i.e., good) path